src/shims/unix/solarish/foreign_items.rs

```diff
@@ -26,21 +26,22 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 // THREAD_NAME_MAX allows a thread name of 31+1 length
                 // https://github.com/illumos/illumos-gate/blob/7671517e13b8123748eda4ef1ee165c6d9dba7fe/usr/src/uts/common/sys/thread.h#L613
                 let max_len = 32;
-                let res = this.pthread_setname_np(
-                    this.read_scalar(thread)?,
-                    this.read_scalar(name)?,
-                    max_len,
-                )?;
+                let res = this
+                    .pthread_setname_np(this.read_scalar(thread)?, this.read_scalar(name)?, max_len)
+                    .map(|r| if r { Scalar::from_u32(0) } else { this.eval_libc("ERANGE") })?;
```

Comment:
Why do you do this with `map`? Just do something like
```rust
let success = pthread_setname_np()?;
let res = if success { ... } else { ... };
```

Reply:
Higher-order functions increase complexity and should only be used when they strictly improve code clarity, which I don't think is the case here. It's also a rather unusual pattern to `map` before `?`, making the reader go "hu, why are they doing this, there must be something non-trivial happening".

Reply:
But isn't `map` on `Result` designed exactly for that?

Reply:
In my understanding it is designed for cases where you need to keep the `Result`. But here we are taking the `Result` apart with `?`.